### DISCUSSION TITLE
Misorientation calculations

### DISCUSSION THREAD

**Discussion by kshankar7** [HEXRD/hexrd#643] [Q&A]
I am trying to find mis-orientation between two orientations (`q1, q2`) defined by two quaternions. My material is defined using a h5 file and is read in (`qsym`). To calculate misorientation I am using the following function call. Can you help me if this is correct? Any help in this would be greatly appreciated.

```
from hexrd.material import Material
from hexrd import rotations  as rot

mat = Material(name='Al_7085', 
               material_file='materials.h5')

qsym = mat.planeData.getQSym()

angle, mis_ori = rot.misorientation(_q1, _q2, ([qsym]))
```
@darrencpagan

**Comment by donald-e-boyce** [comment]
I improved the documentation for rotations.misorientation and created a test file in pull request #644.  

In your example:
```
angle, mis_ori = rot.misorientation(_q1, _q2, ([qsym]))
```
The variable `_q1` should be a single quaternion (shape (4, 1)); `_q2` an array of quaternions (4, n); the third argument should just be (qsym,), which is a 1-tuple.  The results will be `angle`, a scalar array of length `n`, and `mis_ori`, a quaternion array of shape `(4, n)`.

The new documentation looks like:

```
def misorientation(q1, q2, *args):
    """
    PARAMETERS
    ----------
    q1: array(4, 1)
        a single quaternion
    q2: array(4, n)
        array of quaternions
    *args: tuple
        there can be only one extra argument, which is 1- or 2-tuple with
        symmetries (quaternion arrays); for crystal symmetry only,
        use a 1-tuple; with both crystal and sample symmetry use a 2-tuyple

    RETURNS
    -------
    angle: array(n)
        the misorientation angle between `q1` and each quaternion in `q2`
    mis: array(4, n)
        the quaternion of the smallest misorientation angle
    """
```

**Reply by donald-e-boyce** [reply]
One more comment on some python details. 

Starting with your final line.  
```
angle, mis_ori = rot.misorientation(_q1, _q2, ([qsym]))
```

That would work if you added the `*` operator.  
```
angle, mis_ori = rot.misorientation(_q1, _q2, *([qsym]))
```
The python `*args` mechanism takes a list of further arguments.  You can just give them directly without the `*`, or if they are in a list, then the `*` expands the list.  So you could also write:
```
angle, mis_ori = rot.misorientation(_q1, _q2, [qsym])
```
There you are using a list instead of a tuple, which is fine, but if you wanted to use a tuple, it would be:
```
angle, mis_ori = rot.misorientation(_q1, _q2, (qsym,))
```
You need the comma to make it a tuple. Otherwise, it will just be parentheses around a single item that have no effect, i.e. `(qsym)` is the same as `qsym`.
